### PR TITLE
feat: CNS-fvm-update

### PR DESCRIPTION
**Pull request by Yaroms**
# Description

Closes: #XXXX

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] read the [contribution guide](https://github.com/lavanet/lava/blob/main/CONTRIBUTING.md)
* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:
    
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the `main` branch
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration tests
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

File: cookbook/specs/fvm.json [hunk 1]
```diff
@@ -5,21 +5,19 @@
         "specs": [
```
Comment (oren-lava) [inline cookbook/specs/fvm.json:5]: missing APIs: AuthNew, AuthVerify, ChainCheckBlockstore, ChainDeleteObj, ChainExportRangeInternal, ChainHotGC, ChainPrune, ChainPutObj, ChainSetHead, CreateBackup, EthGetTransactionReceiptLimited, GasEstimateFeeCap, GasEstimateGasLimit, GasEstimateGasPremium, GasEstimateMessageGas, ID, LogAlerts, LogList, LogSetLevel

go to https://docs.filecoin.io/reference/json-rpc. The list above is until "Log". There are many missing APIs, add those

**Review by oren-lava — CHANGES_REQUESTED** [review]

File: cookbook/specs/fvm.json [hunk 2]
```diff
@@ -1831,21 +2833,165 @@
                             },
                             {
                               "name": "pruning",
+                              "parse_directive": {
+                                "function_tag": "GET_BLOCK_BY_NUM"
+                                },
                               "values": [
                                 {
-                                  "expected_value": ""
+                                  "latest_distance": 2800
```
Comment (oren-lava) [inline cookbook/specs/fvm.json:2841]: this should be the number of blocks in 24H.
Blocks24H = 24 * 60 * 60 / 30 (block time) = 2880

File: cookbook/specs/fvm.json [hunk 3]
```diff
@@ -1831,21 +2833,165 @@
...
                                 }
                               ]
                             }
+                        ],
+                        "extensions": [
+                            {
+                                "name": "archive",
+                                "cu_multiplier": 5,
+                                "rule": {
+                                    "block": 2600
```
Comment (oren-lava) [inline cookbook/specs/fvm.json:2855]: this should be 23H and 40 min:
Blocks20M = 20 * 60 / 30 = 40
Block23H40M = Blocks24H - Blocks20M = 2840

File: cookbook/specs/fvm.json [hunk 4]
```diff
@@ -1831,21 +2833,165 @@
...
+                                },
                               "values": [
                                 {
-                                  "expected_value": ""
+                                  "latest_distance": 2800
                                 },
                                 {
                                   "extension": "archive",
-                                  "expected_value": ""
+                                  "expected_value": "1"
```
Comment (oren-lava) [inline cookbook/specs/fvm.json:2845]: according to the filecoin docs: https://docs.filecoin.io/networks/mainnet, it began on block 148888, not 1

**Review by oren-lava — CHANGES_REQUESTED** [review]

File: cookbook/specs/fvm.json (hunk 2 above)
Comment (Yaroms) [reply cookbook/specs/fvm.json:2841]: done

File: cookbook/specs/fvm.json (hunk 3 above)
Comment (Yaroms) [reply cookbook/specs/fvm.json:2855]: done

**Review by oren-lava — APPROVED** [review]

**Pull request merged**